src/crepe/api/Scene.h

```diff
@@ -53,6 +58,55 @@ class Scene {
 	//! Mediator reference
 	OptionalRef<Mediator> mediator;
 	//! \}
+
+protected:
+	/**
+	* \brief Retrieve the reference to the SaveManager instance
+	*
+	* \returns A reference to the SaveManager instance held by the Mediator.
+	*/
+	SaveManager & get_save_manager() const { return mediator->save_manager; }
+
+	/**
+	 * \brief Create a new game object using the component manager
+	 *
+	 * \param name Metadata::name (required)
+	 * \param tag Metadata::tag (optional, empty by default)
+	 * \param position Transform::position (optional, origin by default)
+	 * \param rotation Transform::rotation (optional, 0 by default)
+	 * \param scale Transform::scale (optional, 1 by default)
+	 *
+	 * \returns GameObject interface
+	 *
+	 * \note This method automatically assigns a new entity ID
+	 */
+	GameObject new_object(const std::string & name, const std::string & tag = "",
+						  const vec2 & position = {0, 0}, double rotation = 0,
+						  double scale = 1) {
+		// Forward the call to ComponentManager's new_object method
+		return mediator->component_manager->new_object(name, tag, position, rotation, scale);
+	}
+
+	/**
+	 * \brief Mark a resource as persistent (i.e. used across multiple scenes)
+	 *
+	 * \param asset Asset the concrete resource is instantiated from
+	 * \param persistent Whether this resource is persistent (true=keep, false=destroy)
+	 */
+	void set_persistent(const Asset & asset, bool persistent) {
+		mediator->resource_manager->set_persistent(asset, persistent);
+	}
+
+	/**
+	 * \brief Log a message using Log::logf
+	 *
+	 * \tparam Args Log::logf parameters
+	 * \param args  Log::logf parameters
+	 */
+	template <typename... Args>
+	void logf(Args &&... args);
```

Comment:
Deze forwarding werkt helaas niet voor `logf` (hier kwam ik gisteren pas achter op `loek/replay`), het is dit voor de .h header:
```cpp
/**
 * \name Logging functions
 * \see Log
 * \{
 */
//! \copydoc Log::logf
template <class... Args>
void logf(const Log::Level & level, std::format_string<Args...> fmt, Args &&... args);
//! \copydoc Log::logf
template <class... Args>
void logf(std::format_string<Args...> fmt, Args &&... args);
// \}
```

en dit in de .hpp

```cpp
template <class... Args>
void Scene::logf(const Log::Level & level, std::format_string<Args...> fmt, Args &&... args) {
	Log::logf(level, fmt, std::forward<Args>(args)...);
}

template <class... Args>
void Scene::logf(std::format_string<Args...> fmt, Args &&... args) {
	Log::logf(fmt, std::forward<Args>(args)...);
}
```